scripts/core/EXT.rst

```diff
@@ -0,0 +1,108 @@
+<%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+==============
+Extensions
+==============
+
+Objective
+=========
+
+Extensions allow for additional functionality to be added to the Unified Runtime specification
+without affecting the core specification. Extensions may in future be promoted to the core specification
+in a later version of Unified Runtime, when agreed upon by the Working Group. All extensions are 
+optional and are not required to be implemented by any particular adapter, but are expected to 
+be widely available and possibly be required in a future version of the specification. Adapters 
+must report which extensions are supported through the ${x}PlatformGetExtensionProperties query.
+Each extension may also impose additional restrictions on when it can be used - i.e. a platform
+or device query.
+
+There are two types of extensions defined by this specification:
+
+1. **Standard** -  extension will be included into the current and all future version of the specification.
+2. **Experimental** - extensions require additional experimentation and development, before becoming a standard extension. 
+   Applications should not rely on experimental extensions in production.
+
+Requirements
+============
+
+- Extensions must use globally unique names for macros, enums, structures and functions
+- Extensions must have globally unique extension names reported from ${x}PlatformGetExtensionProperties
+- All extensions must be defined in this specification
+- Extensions must not break backwards compatibility of the core APIs
```

Comment:
I think an extension can alter the behavour of an existing API. For example you might want to chain an additional structure to the input or use extension specific enums, but no extension usage should remain the same.

Reply:
Hmm, I'm not sure I'm so confortable with extensions that change semantics because we have no way to opt-out with this architecture. If we were able to enable extensions, like Vulkan, maybe I'd be more conformtable with that.